### PR TITLE
#21898 Sankey Chart: Drag&drop crashes browser - fix

### PR DESCRIPTION
**Pull request by vtkalek**

### CODE REVIEW

File: src/visual.ts [hunk 1]
```diff
@@ -1381,11 +1381,11 @@ module powerbi.extensibility.visual {
                 // Update each link related with this node
                 node.links.forEach( (link: SankeyDiagramLink) => {
                     // select link svg element by ID generated in link creation as Source-Destination
-                    d3.select(`#linkLabelPaths${(link.source.label.name || "").replace(/\W*/g,"")}-${(link.destination.label.name || "").replace(/\W*/g,"")}`).attr({
+                    d3.select(`#linkLabelPaths${('_' + link.source.label.name || "").replace(/\W*/g, "")}-${('_' + link.destination.label.name || "").replace(/\W*/g,"")}`).attr({
```
Comment (zBritva) [inline src/visual.ts]: There is 'linkLabelPaths' as prefix already

File: src/visual.ts [hunk 2]
```diff
@@ -1381,11 +1381,11 @@ module powerbi.extensibility.visual {
...
                         // get updated path params based on actual positions of node
                         d: sankeyVisual.getLinkLabelSvgPath(link)
                     });
-                    d3.select(`#${(link.source.label.name || "").replace(/\W*/g,"")}-${(link.destination.label.name || "").replace(/\W*/g,"")}`).attr({
+                    d3.select(`#${('_' + link.source.label.name || "").replace(/\W*/g, "")}-${('_' + link.destination.label.name || "").replace(/\W*/g,"")}`).attr({
```
Comment (zBritva) [inline src/visual.ts]: There is ES6 pattern. I think the better way is `#_${(link.soource ..... 
And underline at start is enough

File: src/visual.ts [hunk 3]
```diff
@@ -1506,7 +1506,7 @@ module powerbi.extensibility.visual {
                     d: (link: SankeyDiagramLink) => {
                         return this.getSvgPath(link);
                     },
-                    id: (link: SankeyDiagramLink) => `${(link.source.label.name || "").replace(/\W*/g,"")}-${(link.destination.label.name || "").replace(/\W*/g,"")}`
+                    id: (link: SankeyDiagramLink) => `${('_' + link.source.label.name || "").replace(/\W*/g, "")}-${('_' + link.destination.label.name || "").replace(/\W*/g,"")}`
```
Comment (zBritva) [inline src/visual.ts]: `_${(link.source.label.name || "").replace(/\W*/g, "")}-

**Comment by zBritva** [comment]
Changes break the link labels displaying

File: src/visual.ts [hunk 4]
```diff
@@ -1519,6 +1521,9 @@ module powerbi.extensibility.visual {
 
             return linksSelection;
         }
+        public static createLink(link: SankeyDiagramLink, addLinkLabelPath: boolean = false): string {
+                return (addLinkLabelPath ? `#linkLabelPaths` : ``) +  `${('_' + link.source.label.name || "").replace(/\W*/g, "")}-${('_' + link.destination.label.name || "").replace(/\W*/g, "")}`;
```
Comment (zBritva) [inline src/visual.ts]: remove #.  The id of an element must not contain prefix.

File: src/visual.ts [hunk 5]
```diff
@@ -1381,11 +1381,11 @@ module powerbi.extensibility.visual {
                 // Update each link related with this node
                 node.links.forEach( (link: SankeyDiagramLink) => {
                     // select link svg element by ID generated in link creation as Source-Destination
-                    d3.select(`#linkLabelPaths${(link.source.label.name || "").replace(/\W*/g,"")}-${(link.destination.label.name || "").replace(/\W*/g,"")}`).attr({
+                    d3.select(SankeyDiagram.createLink(link, true) ).attr({
```
Comment (zBritva) [inline src/visual.ts]: Change to d3.select(`#` + SankeyDiagram.createLink(link, true) ).attr({
#is required to select by id.

File: src/visual.ts [hunk 6]
```diff
@@ -1587,7 +1594,9 @@ module powerbi.extensibility.visual {
             textPathSelection
                 .attr({
                     startOffset: "50%",
-                    href: (link: SankeyDiagramLink) => `#linkLabelPaths${(link.source.label.name || "").replace(/\W*/g,"")}-${(link.destination.label.name || "").replace(/\W*/g,"")}`
+                    href: (link: SankeyDiagramLink) => {
+                        return SankeyDiagram.createLink(link, true);
```
Comment (zBritva) [inline src/visual.ts]: Change to 
return "#" + SankeyDiagram.createLink(link, true);
#is required to select by id

File: src/visual.ts [hunk 7]
```diff
@@ -1556,7 +1561,9 @@ module powerbi.extensibility.visual {
                     d: (link: SankeyDiagramLink) => {
                         return this.getLinkLabelSvgPath(link);
                     },
-                    id: (link: SankeyDiagramLink) => `linkLabelPaths${(link.source.label.name || "").replace(/\W*/g,"")}-${(link.destination.label.name || "").replace(/\W*/g,"")}`
+                    id: (link: SankeyDiagramLink) => {
+                        return SankeyDiagram.createLink(link);
```
Comment (zBritva) [inline src/visual.ts]: Change to 
return SankeyDiagram.createLink(link, true);
linkLabelPaths prefix is required

**Comment by ignatvilesov** [comment]
Duplicate of #36

**Pull request closed without merging**